pkgs/top-level/all-packages.nix

```diff
@@ -9744,7 +9744,7 @@ with pkgs;
 
   md2gemini = with python3.pkgs; toPythonApplication md2gemini;
 
-  mdbook = callPackage ../tools/text/mdbook {
+  mdbook = callPackage ../by-name/md/mdbook/package.nix {
     inherit (darwin.apple_sdk.frameworks) CoreServices;
   };
```

Comment:
Ah... I am not sure about this part ... with `by-name`, a `callPackage` call is not required anymore, but I am not sure about the inherit here...

Reply:
I gave it a try and it didn't work, but thinking about it i might've messed something up

Reply:
Yeah, the inherit is not working:

```
error:
       … while calling the 'abort' builtin
         at /nix/store/c3qy0g245qam7aqcg6p0p5qc9fgwgaj8-source/lib/customisation.nix:269:13:
          268|        # which is especially relevant with allowAliases = false
          269|        else abort "lib.customisation.callPackageWith: ${error}";
             |             ^
          270|

       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "CoreServices" at /nix/store/c3qy0g245qam7aqcg6p0p5qc9fgwgaj8-source/pkgs/by-name/md/mdbook/package.nix:1'
```

I could modify it but it would not be backwards compatible with `.override`s, which is discouraged by the `pkgs/by-name/README.md`.

Reply:
I am not sure how to fix this.

Reply:
I can probably hack around this https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md#changing-implicit-attribute-defaults, to make it work. I'll give it a try.

Reply:
Seems to work :open_mouth: